fetcher/fetcher_test.go

```diff
@@ -150,12 +149,18 @@ func TestFetchSameKeysSlow(t *testing.T) {
 			stateKeys.Add(strconv.Itoa(k), state.Read)
 		}
 		txID := ids.GenerateTestID()
-		if i%2 == 0 {
-			time.Sleep(1 * time.Second)
-		}
-		require.NoError(f.Fetch(ctx, txID, stateKeys))
-		go func(txID ids.ID) {
+
+		delay := make(chan struct{})
+		go func(txID ids.ID, i int) {
```

Comment:
Usually you just take the scope from outside the goroutine and don't populate these args:

```
go func() {
....
}
```